pkg/cloud/services/networking/router.go

```diff
@@ -185,41 +186,41 @@ func setRouterExternalIPs(client *gophercloud.ServiceClient, openStackCluster *i
 	return nil
 }
 
-func (s *Service) DeleteRouter(openStackCluster *infrav1.OpenStackCluster, network *infrav1.Network) error {
-	if network.Router == nil || network.Router.ID == "" {
-		s.logger.V(4).Info("No need to delete router since no router exists.")
-		return nil
-	}
-	exists, err := s.existsRouter(network.Router.ID)
+func (s *Service) DeleteRouter(openStackCluster *infrav1.OpenStackCluster, clusterName string) error {
+	routerName := getRouterName(clusterName)
+	router, err := s.getRouterByName(routerName)
 	if err != nil {
 		return err
 	}
-	if !exists {
-		s.logger.Info("Skipping router deletion because router doesn't exist", "router", network.Router.ID)
-		return nil
+
+	subnetName := getSubnetName(clusterName)
+	subnet, err := s.getSubnetByName(subnetName)
+	if err != nil {
+		return err
 	}
-	if network.Subnet == nil || network.Subnet.ID == "" {
-		s.logger.V(4).Info("Skipping removing router interface since no subnet exists.")
-	} else {
-		_, err = routers.RemoveInterface(s.client, network.Router.ID, routers.RemoveInterfaceOpts{
-			SubnetID: network.Subnet.ID,
-		}).Extract()
-		if err != nil {
-			if errors.IsNotFound(err) {
-				s.logger.V(4).Info("Router Interface already removed, no actions", "id", network.Router.ID)
-				return nil
+
+	if router.ID != "" {
+		if subnet.ID != "" {
+			_, err = routers.RemoveInterface(s.client, router.ID, routers.RemoveInterfaceOpts{
+				SubnetID: subnet.ID,
+			}).Extract()
+			if err != nil {
+				if capoerrors.IsNotFound(err) {
+					s.logger.V(4).Info("Router Interface already removed, no actions", "id", router.ID)
+					return nil
```

Comment:
when we return here we skip the router deletion. I don't think that's intended

Reply:
Fixed deleting `return nil`